probes/pkg/ping/client/client.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pingclient
+
+import (
+	"flag"
+	"net/http"
+	"time"
+
+	"k8s.io/klog"
+	"k8s.io/perf-tests/probes/pkg/metrics"
+)
+
+var (
+	pingServerAddress = flag.String("ping-server-address", "", "The address of the ping server")
+	pingSleepDuration = flag.Duration("ping-sleep-duration", 1*time.Second, "Duration of the sleep between pings")
+)
+
+// Run runs the ping client probe that periodically pings the ping server and exports latency metric.
+func Run() {
```

Comment:
How about:
- create a pingClientConfig struct
- add a defaultPingClientConfig() method (that will initialize config from flags)
- let Run() take the config as a parameter

I think it would be a bit cleaner.
And same for server below.